pkg/admission/pathannotation/pathannotation_admission_test.go

```diff
@@ -0,0 +1,313 @@
+/*
+Copyright 2023 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pathannotation
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/kcp-dev/logicalcluster/v3"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/admission"
+	"k8s.io/apiserver/pkg/endpoints/request"
+
+	apisv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/apis/v1alpha1"
+	"github.com/kcp-dev/kcp/pkg/apis/core"
+	corev1alpha1 "github.com/kcp-dev/kcp/pkg/apis/core/v1alpha1"
+	schedulingv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/scheduling/v1alpha1"
+	tenancyv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/tenancy/v1alpha1"
+)
+
+var wellKnownAdmissionContext = request.WithCluster(context.Background(), request.Cluster{Name: "test-cluster"})
+
+func TestPathAnnotationAdmit(t *testing.T) {
+	scenarios := []struct {
+		name              string
+		admissionObject   runtime.Object
+		admissionResource schema.GroupVersionResource
+		admissionVerb     admission.Operation
+		admissionOptions  runtime.Object
+		admissionContext  context.Context
+		getLogicalCluster func(clusterName logicalcluster.Name, name string) (*corev1alpha1.LogicalCluster, error)
+
+		expectError             bool
+		validateAdmissionObject func(t *testing.T, obj runtime.Object)
+	}{
+		{
+			name:             "error when no cluster in the context",
+			admissionContext: context.TODO(),
+			expectError:      true,
+		},
+		{
+			name:              "admission is not applied to logicalclusters", // would panic since no admission obj
```

Comment:
What does the comment mean? It _would_ panic, but it doesn't, and why. ..?

Reply:
I was lazy - instead of providing an object and making sure it doesn't have the annotation set i assumed it would panic if the admission would apply. Anyway I added an object and provided `objectWithoutPathAnnotation` function.